chains/orchestrator-paras/node/src/service.rs

```diff
@@ -566,9 +568,20 @@ fn start_consensus_orchestrator(
             let relay_chain_interface = relay_chain_interace_for_cidp.clone();
             let client_set_aside_for_cidp = client_set_aside_for_cidp.clone();
             async move {
-                let para_ids = client_set_aside_for_cidp
+                // We added a new runtime api that allows to know which parachains have
+                // some collators assigned to them. We'll now only include those. For older
+                // runtimes we continue to write all of them.
```

Comment:
Can someone explain what does this mean? (and add to PR description). Do we include the headers for all parachains even if the block number has not changed? Or is this only an optimization to not have to check all the headers, but the created inherent data will be the same with and without the runtime api?

Reply:
To know that the block number didn't change you have to take a look at the relay, which is what the proof allows. We know that if the parachain don't have assigned collators, new blocks won't be produced, so we can skip that and include a smaller amount of proofs.

This PR update both the runtime part that verifies all the chains have their proof (now only the ones with assigned collators), and the client part that will now only insert the required proofs.

Reply:
Ok thanks! I was missing the fact that even if the block number does not change, we still need to include the header in the inherent to prove that it didn't change.